Declare a cipher unsupported for direct AD integration (#3549)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-the-active-directory-authentication-source-on-projectserver.adoc

Enable Active Directory (AD) users to access {Project} by configuring the corresponding authentication provider on your {ProjectServer}.

[NOTE]
====

> {Project} does not support using the `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384` cipher suite.

Reviewer: I'm not sure we should call it "does not support". Perhaps something like:
```suggestion
The `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384` cipher suite is known to cause incompatibilities.
```

<html><body>You are being <a href="{foreman-example-com}/hosts">redirected</a>.</body></html>
----

.Troubleshooting
* Connecting to the AD LDAP can sometimes fail with an error such as the following:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> { "error": { "message": "ERF77-7629 [Foreman::LdapException]: Error while connecting to 'server.com' LDAP server at 'ldap.example.com' during authentication ([Net::LDAP::Error]: Connection reset by peer - SSL_connect)" } }

Reviewer: Technically this is API output. I think the `message` part can also be seen in the UI. Probably if you use the `Test connection` button if it's 100% reproducible but I don't have the setup to test this.

Author: What if I clarify that the error message appears in the logs? That's how the issue was troubleshooted. Does it make sense?

Reviewer 2: > if it's 100% reproducible

Sadly it's not, the failure is intermittent. From what I've seen I'd say each ldap lookup has a 1:50 chance of hitting this.

> What if I clarify that the error message appears in the logs?

That would make sense to me

----
+
If you see this error, verify which cipher is used for the connection:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# openssl s_client -connect _ldap.example.com_:636
----
+
If the `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384` cipher is used, disable it on either the {ProjectServer} side or on the AD side.
The `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384` cipher is known to cause incompatibilities.
+
ifdef::satellite[]
For more information, see the Red{nbsp}Hat Knowledgebase solution link:https://access.redhat.com/solutions/4870221[API calls to Red Hat Satellite 6 fail intermittently on LDAP authentication].
endif::[]

> ifndef::orcharhino[]

Reviewer: Perhaps limit this to EL-based installs?
```suggestion
ifdef::foreman-el,katello[]
```
